user_linux_cgo.go

```diff
@@ -0,0 +1,51 @@
+// +build cgo,linux
+
+package buildah
+
+// #include <pwd.h>
+// #include <stdlib.h>
+// #include <stdio.h>
+// #include <string.h>
+import "C"
+
+import (
+	"path/filepath"
+	"sync"
+	"unsafe"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func getUser(rootdir, username string) (specs.User, error) {
```

Comment:
I empathize.  For handling the equivalent of a RUN instruction while building an image, though, I think it'll have to be sufficient.  It mirrors what we get from `docker build` when evaluating a Dockerfile, and we're trying to avoid surprising people who are used to that.